drivers/adc/Kconfig

```diff
@@ -33,6 +33,11 @@ config ADC_CONFIGURABLE_INPUTS
 config ADC_CONFIGURABLE_EXCITATION_CURRENT_SOURCE_PIN
 	bool
 
+# By selecting or not this option particular ADC drivers indicate if they
+# allow a configurable voltage bias pin.
+config ADC_CONFIGURABLE_VBIAS_PIN
```

Comment:
why not in DT? if it describes HW?

Reply:
This only enables the feature in software and increases the size of `struct adc_channel_cfg`, similar to `ADC_CONFIGURABLE_INPUTS` and `ADC_CONFIGURABLE_EXCITATION_CURRENT_SOURCE_PIN`.

Reply:
yes, I don't like adc_configurable_inputs either

with kconfig alone there cannot be instance-based determination of the feature support

Reply:
Wouldn't this then require the extra members to always be present in the `struct adc_channel_cfg`? I suppose some people then might complain about the footprint. Probably the reason why it is like that now.

But besides this, having some consistency is important in my opinion. And as long `ADC_CONFIGURABLE_INPUTS` is implemented like this, I wouldn't deviate from the approach in the same subsystem for something very similar.

Reply:
> Wouldn't this then require the extra members to always be present in the `struct adc_channel_cfg`? I suppose some people then might complain about the footprint. Probably the reason why it is like that now.
> 
> But besides this, having some consistency is important in my opinion. And as long `ADC_CONFIGURABLE_INPUTS` is implemented like this, I wouldn't deviate from the approach in the same subsystem for something very similar.

I talked to @anangl about this before and he told me there is not a reason why this is a kconfig https://discord.com/channels/720317445772017664/917734331135438878/1096019869666918400

I think some DT solution for this should be considered

Reply:
Okay, so the plan is to change this, but it didn't happen yet? I don't think a PR which adds a new feature should change such a fundamental thing, it should rather be consistent with the already existing features.
Are there any plans on changing this soon?

Reply:
I am not suggesting you to change the configurable inputs

Reply:
But implementing the voltage biasing differently than the configurable inputs is a rather nasty inconsistency in the approach, in my opinion. Something I would like to avoid.

Besides this, in my use case I must be able to select the usage of one of these features during runtime, as I have either a AD590, a thermocouple or a PT100 connected to the ADC. Maybe you could illustrate how you would accommodate such a use case with a device tree approach? Because I'm not yet completely sure where you want to head to with moving these options into the devicetree.

Reply:
for now, the Kconfig is fine

Reply:
for now the Kconfig alone is fine, but I just have one question which is, is there any place for a generic property related to this in the adc-controller binding. See how for example the configurable inputs has channel properties in the binding

Reply:
done